hub/core/dataset/dataset.py

```diff
@@ -532,17 +534,21 @@ def _get_total_meta(self):
 
     def _set_derived_attributes(self):
         """Sets derived attributes during init and unpickling."""
-
         if self.index.is_trivial() and self._is_root():
             self.storage.autoflush = True
 
         if not self.version_state:
             self._load_version_info()
 
         self._populate_meta()  # TODO: use the same scheme as `load_info`
-        self.info = load_info(get_dataset_info_key(self.version_state["commit_id"]), self.storage, self.version_state)  # type: ignore
         self.index.validate(self.num_samples)
 
+    @property
+    def info(self):
+        if self._info is None:
```

Comment:
If `ds.info` is updated, how will the update be picked up?

Reply:
info uses callbacks, so just updating ds.info will update in place. If you're talking about some other user updating it in the background, that shouldn't be possible due to locking.